### PR TITLE
add vm testcases

### PR DESCRIPTION
**Pull request by AhmedHanafy725**
- Test case for creating a vm with ubuntue and zero-os operating system.
- Create vm with type default.
- Add and delete disk to the vm.
- Add server on vm port.
- Add and delete zerotier network to vm.
- Enable and disable the vnc port on the vm.
- Pause and resume the vm with type default.
- Reset and reboot the vm.
- Shutdown and start vm.
- edit execute_command method
- edit load_ssh_key method
vmtest
screenshot from 2018-09-11 17-34-16

### CODE REVIEW

File: tests/integration_tests/testcases/base_test.py [hunk 1]
```diff
@@ -182,3 +182,16 @@ def get_gw_network(self, gw, name):
         network = [network for network in networks if network.name == name ]
         return network
 
+    def is_completely_shutdown(self, created_vm):
```
Comment (Dinaamagdy) [inline tests/integration_tests/testcases/base_test.py]: make it generic by make it check status according to passing status

File: tests/integration_tests/testcases/base_test.py [hunk 2]
```diff
@@ -149,17 +149,17 @@ def host_join_zt(self, zt_network=None):
         host_ip = host_member.private_ip
         return host_ip 
 
-    def ssh_vm_execute_command(self, vm_ip, cmd):
+    def ssh_vm_execute_command(self, vm_ip, cmd, port=22):
         for _ in range(10):            
-            result, error = self.execute_command(ip=vm_ip, cmd=cmd)
-            if error:
-                print(colored(' [-] Execute command error : {}'.format(error), 'red'))
+            resposne = self.execute_command(ip=vm_ip, cmd=cmd, port=port)
```
Comment (Dinaamagdy) [inline tests/integration_tests/testcases/base_test.py]: it was more clear when it return error and result

File: tests/integration_tests/testcases/base_test.py [hunk 3]
```diff
@@ -182,3 +182,16 @@ def get_gw_network(self, gw, name):
...
+        for _ in range(200):
+            if not created_vm.is_running():
+                return
+            else:
+                time.sleep(1)
+        else:
+            self.assertFalse(created_vm.is_running(),'Take long time to shutdown')
+
+    def check_vnc_connection(self, vnc_ip_port):
+        vnc = 'vncdotool -s {} type {} key enter'.format(vnc_ip_port, repr('ls'))
```
Comment (Dinaamagdy) [inline tests/integration_tests/testcases/base_test.py]: check that vncdotool added on requirement file

Comment (AhmedHanafy725) [reply tests/integration_tests/testcases/base_test.py]: already i did

File: tests/integration_tests/testcases/test01_vms.py [hunk 4]
```diff
@@ -26,47 +29,139 @@ def test001_create_vm(self):
...
 
         self.log("Check that vm added to zerotier network and can access it using it, should succeed.")
-        ztIdentity = self.vm.data["ztIdentity"]
+        ztIdentity = vm.data["ztIdentity"]
         vm_zt_ip = self.get_machine_zerotier_ip(ztIdentity)
         result = self.ssh_vm_execute_command(vm_ip=vm_zt_ip, cmd='pwd')
-        self.assertNotEqual(len(result), 0)
+        self.assertEqual(result, '/root')
+
+        self.log("Destroy vm [VM1], should succeed.")
```
Comment (Dinaamagdy) [inline tests/integration_tests/testcases/test01_vms.py]: ad destroying the vm on teardown too , to not make it every testcase

File: tests/integration_tests/testcases/test01_vms.py [hunk 5]
```diff
@@ -177,25 +433,71 @@ def test002_pause_and_resume_vm(self):
...
+            self.skipTest('https://github.com/threefoldtech/jumpscale_lib/issues/102')
+
+        self.log("Create a vm[vm1], should succeed.")
+        self.create_booted_vm(os_type)
+
+        self.log("Pause [vm1], should succeed.")
+        self.vm.pause()
+
+        self.log("Check that [vm1] has been paused successfully.")
+        result1 = self.execute_command(ip=self.vm_zt_ip, cmd='pwd')
```
Comment (Dinaamagdy) [inline tests/integration_tests/testcases/test01_vms.py]: check pause  status too

File: tests/integration_tests/testcases/test01_vms.py [hunk 6]
```diff
@@ -177,25 +433,71 @@ def test002_pause_and_resume_vm(self):
...
         #. Reset or reboot the vm, should suceeed.
+        #. Reboot/reset the VM
         #. Check that [vm] has been rebooted/reset successfully.
-        """
-        pass
+        """       
+        self.log("Create a vm[vm1], should succeed.")
+        self.create_booted_vm(os_type)
+
+        self.log("Reboot the VM")
```
Comment (Dinaamagdy) [inline tests/integration_tests/testcases/test01_vms.py]: you can parametize reboot and reset too , to avoid duplicated code

**Pull request merged**